### PR TITLE
add adadelta

### PR DESCRIPTION
**Pull request by dx111**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Checklist
**Before PR**:

- [x] I have read and followed the workflow indicated in the [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: impl/camb/functions/adadelta.cpp [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                                     squareAvgCasted.data(),
+                                     accDeltaDesc.get(),
+                                     accDeltaCasted.data(),
+                                     gradDesc.get(),
+                                     gradCasted.data(),
+                                     lrTensor.data(),
+                                     rhoTensor.data(),
+                                     epsTensor.data()));
+
+    DIOPI_CALL(dataTypeCast(ctx, inputTensor, inputCasted));
```
Comment (yangbofun) [inline impl/camb/functions/adadelta.cpp:64]: 这里检查下，是不是遗漏了squareAvgCasted、 accDeltaCasted的回写。

File: adaptor/codegen/gen.py [hunk 2]
```diff
@@ -62,7 +62,8 @@
 exclude_ops = ['CopyInp', 'CastDtype']
 inp_config = {
     'BatchNorm': ['running_mean', 'running_var'],
-    'IndexPut': ['out']
+    'IndexPut': ['out'],
+    'Adadelta': ['input', 'grad', 'square_avg', 'acc_delta'],
```
Comment (yangbofun) [inline adaptor/codegen/gen.py:66]: 为啥有input

Comment (yewentao256) [reply adaptor/codegen/gen.py:66]: 这里input指的是parameter，是要被更新的

File: impl/camb/functions/adadelta.cpp (hunk 1 above)
Comment (yewentao256) [reply impl/camb/functions/adadelta.cpp:60]: 已经补充上了

**Review by yangbofun — APPROVED** [review]

**Pull request merged**